### PR TITLE
Combine discovery reboot and reboot_all test

### PR DESCRIPTION
**Pull request by shubhamsg199**
Combining discovery reboot and reboot_all test into one

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts
provisioning: true

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8871
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts --external-logging --include-onprem-provisioning
Test Result : ==== 1 failed, 1 passed, 13 deselected, 140 warnings in 2130.18s (0:35:30) =====
```

File: tests/foreman/api/test_discoveredhost.py [hunk 1]
```diff
@@ -353,29 +353,32 @@ def test_positive_refresh_facts_pxe_host(self, module_target_sat):
         """
 
     @pytest.mark.on_premises_provisioning
+    @pytest.mark.parametrize('endpoint', ['reboot', 'reboot_all'])
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_discoveredhost.py:356]: Instead of making it parametrized, I'd suggest to keep it as a separate test

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts
provisioning: true

File: tests/foreman/api/test_discoveredhost.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_discoveredhost.py:356]: Any specific reason for separate test, both would do the same things?

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8874
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts --external-logging --include-onprem-provisioning
Test Result : ==== 1 failed, 1 passed, 13 deselected, 141 warnings in 2000.15s (0:33:20) =====
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts
provisioning: true

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8895
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts --external-logging --include-onprem-provisioning
Test Result : ========== 1 passed, 13 deselected, 30 warnings in 1597.76s (0:26:37) ==========
```

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_pxe_hosts
provisioning: true

File: tests/foreman/api/test_discoveredhost.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_discoveredhost.py]: Like if first endpoint fails for some reason, then second endpoint will be untested, so I'd suggest to keep as it is in separate tests

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_all_pxe_hosts
provisioning: true

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8901
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_discoveredhost.py -k test_positive_reboot_all_pxe_hosts --external-logging --include-onprem-provisioning
Test Result : ========== 1 passed, 14 deselected, 31 warnings in 1614.44s (0:26:54) ==========
```

File: tests/foreman/api/test_discoveredhost.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_discoveredhost.py]: Updated, it was causing some issues. Separate two tests

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**